data/mapping/mapping.json

```diff
@@ -4,31 +4,25 @@
       "analyzer":{
         "gp_name_analyzer":{
           "type":"custom",
+          "tokenizer":"name_tokenizer",
+          "filter":[ "lowercase", "gp_title_stop" ]
+        },
+        "surgery_name_analyzer":{
+          "type":"custom",
           "tokenizer":"standard",
-          "filter":[
-            "lowercase",
-            "gp_stop"
-          ]
+          "filter":[ "lowercase" ]
+        }
+      },
+      "tokenizer": {
+        "name_tokenizer": {
+          "type": "pattern",
+          "pattern": "[^a-zA-Z']+"
         }
       },
       "filter":{
-        "gp_stop":{
+        "gp_title_stop":{
           "type":"stop",
-          "stopwords": [ 
-            "and",
-            "the",
-            "&",
-            "doctor",
-            "doctors",
-            "dr",
-            "drs",
-            "partners",
-            "practice",
-            "surgery",
-            "medical",
-            "health",
-            "centre",
-            "center" ]
+          "stopwords": [ "doctor", "doctors", "dr", "drs" ]
```

Comment:
I'm trying to stick to the clear definition of a stop word in that they are words which occur in so many documents as to make them useless as a search term. I'd argue that handling _general_ and _practitioner_ (as well as the various spelling permutations of professor) is probably a data cleansing exercise to create a proper 'title' property.